libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -125,40 +127,53 @@ public UserIdentity generateUserIdentity(
             String userId, String sub, String vot, ContraIndicators contraIndicators)
             throws HttpResponseExceptionWithErrorBody, CredentialParseException,
                     UnrecognisedCiException {
-        List<VcStoreItem> vcStoreItems = dataStore.getItems(userId);
-
+        ProfileType profileType = Vot.valueOf(vot).getProfileType();
+        List<VcStoreItem> vcStoreItems =
+                VcHelper.filterVCBasedOnProfileType(dataStore.getItems(userId), profileType);
         List<String> vcJwts = vcStoreItems.stream().map(VcStoreItem::getCredential).toList();
 
         String vtm = configService.getSsmParameter(CORE_VTM_CLAIM);
 
         UserIdentity.UserIdentityBuilder userIdentityBuilder =
                 UserIdentity.builder().vcs(vcJwts).sub(sub).vot(vot).vtm(vtm);
 
-        if (Vot.P2.name().equals(vot)) {
-            final List<VcStoreItem> successfulVCStoreItems =
-                    getSuccessfulVCStoreItems(vcStoreItems);
+        buildUserIdentityBasedOnProfileType(
+                vot, contraIndicators, profileType, vcStoreItems, userIdentityBuilder);
+
+        return userIdentityBuilder.build();
+    }
+
+    private void buildUserIdentityBasedOnProfileType(
+            String vot,
+            ContraIndicators contraIndicators,
+            ProfileType profileType,
+            List<VcStoreItem> vcStoreItems,
+            UserIdentity.UserIdentityBuilder userIdentityBuilder)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody {
+        final List<VcStoreItem> successfulVCStoreItems = getSuccessfulVCStoreItems(vcStoreItems);
+        if (!vot.equals(Vot.P2.name()) && profileType.equals(ProfileType.GPG45)) {
+            userIdentityBuilder.returnCode(getFailReturnCode(contraIndicators));
+        } else {
```

Comment:
I think it would be cleaner to move the condition out of this method - so this method is more like `addUserIdentityClaims` and it's a bit more single-responsibility:
```
if (vot.equals(Vot.P0)) {
  userIdentityBuilder.returnCode(getFailReturnCode(contraIndicators));
} else {
  addUserIdentityClaims(userIdentityBuilder, vcStoreItems);
  userIdentityBuilder.returnCode(getSuccessReturnCode(contraIndicators));
}
```
WDYT?